src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py

```diff
@@ -197,6 +204,15 @@ def schema_expression(data_type: DataType, is_nullable: bool) -> str:
         return "to_geography('POINT(-122.35 37.55)')"
     if isinstance(data_type, GeometryType):
         return "to_geometry('POINT(-122.35 37.55)')"
+    if isinstance(data_type, VectorType):
+        make_float = 0.1 if data_type.element_type == "float" else 0
```

Comment:
Nit: This might read a bit more cleanly if the element type check is hoisted to the top of the block and the return logic is deduplicated
```
if data_type.element_type == "int":
    zero = int(0)
elif data_type.element_type == "float":
    zero = float(0)
else:
    raise TypeError(f"Invalid vector element type: {data_type.element_type}")
values = [i + zero for i in range(data_type.dimension]
return f"{values} :: VECTOR({data_type.element_type},{data_type.dimension})"
```